docs/querying/basics.md

```diff
@@ -55,10 +55,19 @@ Example:
 
 ### Float literals
 
-Scalar float values can be literally written as numbers of the form
-`[-](digits)[.(digits)]`.
+Scalar float values can be literally written as integer or floating-point numbers, for example:
 
+    23
     -2.43
+    3.4e-9
+    0x8f
+    -Inf
+
+Prometheus first attempts to parse scalar literals using Go's [`strconv.ParseInt()`](https://golang.org/pkg/strconv/#ParseInt)
```

Comment:
I don't think we should be mentioning implementation details like this, also it's the PromQL grammar that defines this - not the Go implementation.